widgets/hwmontemp_linux.lua

```diff
@@ -25,30 +25,45 @@ local io = { open = io.open }
 local helpers = require"vicious.helpers"
 local spawn = require"vicious.spawn"
 
+local pathcache = {}
+
+local function read_sensor(path)
+    local f = io.open(path, "r")
+    local v = tonumber(f:read"*line") / 1000
+    f:close()
+
+    return v
```

Comment:
I think passing callback here would allow us to avoid, ehm, this.

Reply:
I'm not entirely sure if I understand what you mean. Getting rid of `local v` and calling the passed callback inside of `read_sensor()`?

So that would save us the variable at the cost of an additional parameter that has to be passed. Well, I guess I'm too much of a C guy, trusting the compiler to optimize at this level ;)

Reply:
It was not exactly about (micro-)optimization, I just don't like the *let me save a value as some variable and do something and return it later* thing.  I don't feel really strong about this though so if I can't change your mind, feel free to resolve the conversation and I'll merge.

Reply:
Okay I understand. I'll come up with a code proposal and force-push this pull request branch.

It's going to be at least tomorrow as I'll have no internet access for the rest of today.